devtools/server/actors/replay/connection.js

```diff
@@ -533,6 +527,114 @@ class Recording extends EventEmitter {
   }
 }
 
+function checkShouldValidateUrl() {
+  if (gShouldValidateUrl !== null) {
+    return Promise.resolve(gShouldValidateUrl)
+  } else {
+    return queryAPIServer(`
+      query GetOrgs {
+        viewer {
+          workspaces {
+            edges {
+              node {
+                isOrganization
+                settings {
+                  features
+                }
+              }
+            }
+          }
+        }
+      }
+    `).then(resp => {
+      if (resp.errors) {
+        throw new Error("Unexpected error checking Replay user permissions");
+      }
+
+      const workspaces = resp.data.viewer.workspaces.edges;
+      gShouldValidateUrl = !workspaces ? false : workspaces.some(w => {
+        if (w.node.isOrganization) {
+          const {allowList, blockList} = w.node.settings?.features?.recording || {};
+
+          return (Array.isArray(allowList) && allowList.length > 0) || (Array.isArray(blockList) && blockList.length > 0);
+        }
+      });
+
+      return gShouldValidateUrl;
+    });
+  }
+}
+
+async function canRecordUrl(url) {
+  try {
+    const shouldValidate = await checkShouldValidateUrl();
+    if (!shouldValidate) return true;
+
+    const resp = await queryAPIServer(`
+      query CanRecord ($url: String!) {
+        viewer {
+          canRecordUrl(url: $url)
+        }
+      }
+    `, {
+      url
+    });
+
+    if (resp.errors) {
+      throw new Error(resp.errors[0].message);
+    }
+
+    return resp.data.viewer.canRecordUrl;
+  } catch (e) {
+    // Fallback to allowing recordings if the backend errors but log to telemetry
+    console.error(e);
+    pingTelemetry("recording", "can-record-failed", { why: e.message || "", url });
+
+    return true;
+  }
+}
+
+function getLocationListener(key) {
+  return {
+    key,
+    onLocationChange(aWebProgress, _, aLocation) {
+      if(!aWebProgress.isTopLevel) return;
+
+      // Always allow blank and new tab
+      if (aLocation.displaySpec === "about:blank" || aLocation.displaySpec === "https://app.replay.io/browser/new-tab") {
```

Comment:
Since we set this for the user, it's possible for them to be blocked on starting a recording from a new tab. Perhaps we can query for the replay-specific URL rather than hard code it.

Reply:
Isn't blocking that kind of the point though? I guess I'm saying I expected that to fail unless we included this URL in the `allowList` value explicitly, since recording a random replay page probably isn't something users want.

Reply:
I suppose but the UX is just bad, imo. We don't show the URL in this case so the user will hit record and then get a dialog saying they can't record a thing they didn't know they were trying to record. They'd at least see the URL in the notification but I don't think they can copy/paste it from there so it'll be frustrating.

Reply:
Fair enough. If we want to leave it for now we can always change it later. Is there a constant or a pref for this we can read at least?

Reply:
We can grab the pref for it but it's conceivable that the user could set their own pref to an otherwise disallowed site and we'd let it pass. There isn't a constant for it anywhere atm.

Reply:
Alright happy to leave it as-is for now then.